autoload/ale/sign.vim

```diff
@@ -67,10 +67,14 @@ endfunction
 " Given a loclist, combine the loclist into a list of signs such that only
 " one sign appears per line. Error lines will take precedence.
 " The loclist will have been previously sorted.
-function! ale#sign#CombineSigns(loclist) abort
+function! ale#sign#CombineSigns(buffer, loclist) abort
     let l:signlist = []
 
     for l:obj in a:loclist
+        if has_key(l:obj, 'bufnr') && l:obj.bufnr >= 0 && l:obj.bufnr != a:buffer
```

Comment:
I think the safest option is to not output warnings or errors file files other than the file you are checking for now. Supporting errors for buffers other than the one you are working on requires some serious consideration. Will it break the echo line as the cursor moves? What effect will it have on the warning/error count? Will it break the commands for jumping between warnings/errors?

Remove the warning and errors for other buffers for now, and we can develop that at another time. I think we should get something working for checking for some simple Go errors first, which doesn't suffer from module import problems, and then think about showing more warnings/errors later.